### PR TITLE
Ana/qa august

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Lots of small fixes. 

- Links for validators in TransactionItem not working

- Cannot return null of Delegation.id

image

- checkAddressRole not getting `address` in DelegationModal

- validator addresses in TransactionItem (making extension quite big)

image

- disable voting for Council proposals (only council members can vote). This is the easy solution. A better solution is to query to know if the current address is a council member and can therefore vote

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f4d295635dfbb6a8ca681aa--lunieio.netlify.app

File: app/src/components/transactions/message-view/StakeTxDetails.vue [hunk 1]
```diff
@@ -4,14 +4,26 @@
...
         <img
           v-if="validator && validator.picture"
           :src="validator.picture"
           class="validator-image"
           :alt="`validator logo for ` + validator.name"
         />
-        {{ transaction.details.to[0] | resolveValidatorName(validators) }}
+        {{
+          transaction.details.to[0]
+            | formatAddress
```
Comment (faboweb) [inline app/src/components/transactions/message-view/StakeTxDetails.vue:24]: this looks weird. `resolveValidatorName` already does a `formatAddress`

File: app/src/components/transactions/message-view/UnstakeTxDetails.vue [hunk 2]
```diff
@@ -4,14 +4,26 @@
...
         <img
           v-if="validator && validator.picture"
           :src="validator.picture"
           class="validator-image"
           :alt="`validator logo for ` + validator.name"
         />
-        {{ transaction.details.from[0] | resolveValidatorName(validators) }}
+        {{
+          transaction.details.from[0]
+            | formatAddress
```
Comment (faboweb) [inline app/src/components/transactions/message-view/UnstakeTxDetails.vue:24]: see above

File: app/src/components/transactions/message-view/StakeTxDetails.vue (hunk 1 above)
Comment (Bitcoinera) [reply app/src/components/transactions/message-view/StakeTxDetails.vue:24]: true, I am going to check that

**Comment by faboweb** [comment]
changelogs missing

**Comment by faboweb** [comment]
needs linting

File: app/src/components/transactions/message-view/StakeTxDetails.vue (hunk 1 above)
Comment (Bitcoinera) [reply app/src/components/transactions/message-view/StakeTxDetails.vue]: ok, fixed

**Comment by Bitcoinera** [comment]
OK, all comments applied! Let's merge! :woman_dancing:

File: app/src/filters/index.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ export const formatAddress = (address, length = 4) => {
 }
 
 export const resolveValidatorName = (address, validators) => {
-  if (validators[address]) {
+  if (validators[address] && validators[address].name.length < 20) {
```
Comment (faboweb) [inline app/src/filters/index.js:48]: why is this needed?

Comment (Bitcoinera) [reply app/src/filters/index.js:48]: Because names in Substrate can be the operatorAddress, being super long and thus breaking the extension styles

Comment (faboweb) [reply app/src/filters/index.js:48]: alright! please leave comment. this is not at all obvious

**Comment by faboweb** [comment]
unit tests failing

**Comment by Bitcoinera** [comment]
> unit tests failing + e2e failing

Strangely E2E succeed locally. Let's wait for the next result

image

File: app/src/filters/index.js (hunk 3 above)
Comment (Bitcoinera) [reply app/src/filters/index.js:48]: oh true! Good catch. I thought I had

**Comment by Bitcoinera** [comment]
Maybe GitHub is having some sort of connection problem? The error in the E2E is always different, not consistent at all... Let's see the next one

**Comment by Bitcoinera** [comment]
They passed once! Here is proof :woman_dancing: 

image

Let's merge :rocket:

**Comment by Bitcoinera** [comment]
The program just stops running all of the sudden :shrug:

https://automate.browserstack.com/builds/dd83badf5f105f3d306e614d796308adb4b5d2dd/sessions/6f5110423b7254e9172a591769775b7b55d122e3

**Comment by faboweb** [comment]
> The program just stops running all of the sudden 🤷

just cancel the GitHub Action and rerun next time
